Fearure/issue 220 adpater s3 azure

Usamavirk444 opened the pull request.

Usamavirk444 committed:
feature-issue: add adapter for s3 and azure

Usamavirk444 committed:
featur/issue: fix on the psot according to adapter

Usamavirk444 committed:
feature/issue:adpter move to servies folder and make variables in seprate file and make a common file

usama-tariq1 commented on src/components/Dashboard/posts/create-post-form.tsx:
```diff
@@ -230,17 +239,23 @@ const CreatePostForm: React.FC<Props> = ({ variant = "posts" }) => {
-                newPost.content
-              )
-        if (post && post.data && post.data) {
+                newPost.content,
```
let use named variables so we dont have pass undefined 
like an object
`
createFilePost({
 fileName: "",
fileSize: ""
})
`

usama-tariq1 commented on src/components/Dashboard/posts/create-post-form.tsx:
```diff
@@ -219,6 +219,15 @@ const CreatePostForm: React.FC<Props> = ({ variant = "posts" }) => {
+            : "/posts";
+
         const post =
           variant === "spaces"
```
move this ternary condition to if else or switch case will be easy to understand

usama-tariq1 commented on src/server-actions/FileSharing/FileSharing.ts:
```diff
@@ -23,31 +23,28 @@ export const CreateNewFolderAction = CreateServerAction(true, async(
       fileName,
       fileSize,
-      uploadedFileData[0].id
+      fileRecord[0].id
```
return only one file record from uploadFileAndSaveMetadata

usama-tariq1 commented on src/services/storage/Adapters/azure.adapter.ts:
```diff
@@ -0,0 +1,26 @@
+
+export const AzureStorageAdapter: StorageAdapter = {
+  async uploadFile(fileBuffer, fileName, mimeType, folderPath = "/") {
+    const blobName = `${randomUUID()}-${folderPath +fileName}`;
```
folderPath should be before UUID

usama-tariq1 commented on src/services/storage/types/interface.ts:
```diff
@@ -0,0 +1,8 @@
+export interface StorageAdapter {
+  uploadFile(
+    fileBuffer: Buffer,
+    fileName: string,
```
change to named variable , detail added above

Usamavirk444 committed:
issue -pr-comment-fxies:return only one file record from uploadFileAndSaveMetadata
folderPath should be before UUID

usama-tariq1 commented on src/components/Dashboard/posts/create-post-form.tsx:
```diff
@@ -219,28 +219,50 @@ const CreatePostForm: React.FC<Props> = ({ variant = "posts" }) => {
+
+        if (variant === "spaces") {
+          post = await createFilePost({
+            type: newPost.type,
```
prepare the this object in if else then call the function once

usama-tariq1 commented on src/components/Dashboard/posts/create-post-form.tsx:
```diff
@@ -219,28 +219,50 @@ const CreatePostForm: React.FC<Props> = ({ variant = "posts" }) => {
+          });
+        } else {
+          post = await createFilePost({
+            type: newPost.type,
```
prepare the object in condition and call the the function once after condition

usama-tariq1 commented on src/server-actions/FileSharing/FileSharing.ts:
```diff
@@ -23,31 +23,28 @@ export const CreateNewFolderAction = CreateServerAction(true, async(
-      fileType as string,
-      "/spaces"
-    )
+    const fileBuffer = Buffer.from(fileB64string.split(",")[1], "base64")
```
create this as helper util function

usama-tariq1 commented on src/server-actions/Post/Post.ts:
```diff
@@ -53,62 +53,75 @@ export const CreatePostAction = CreateServerAction(
+export const CreateFilePostAction = CreateServerAction(true, async (args: {
+  type: string;
+  fileSize: number;
+  fileName: string;
```
move these prams args to interface

usama-tariq1 commented on src/services/storage/Adapters/azure.adapter.ts:
```diff
@@ -0,0 +1,36 @@
+  async uploadFile({
+  fileBuffer,
+  fileName,
+  mimeType,
```
use interface for type args

usama-tariq1 commented on src/services/storage/types/interface.ts:
```diff
@@ -0,0 +1,8 @@
+export interface StorageAdapter {
+  uploadFile(params: {
+    fileBuffer: Buffer;
```
use separate interface for params and use it here

Usamavirk444 committed:
feature/issue: fix github comment: use separate interface for params and use it here
use interface for type args
move these prams args to interface
create this as helper util function
prepare the object in condition and call the the function once after condition

Commit:
fix(issue-220): make storage client separate

Usamavirk444 committed:
feature/issue: fix the folder path in the szzure and s3

Usamavirk444 committed:
add azure variable in the next file

Usamavirk444 committed:
Merge branch 'development' of https://github.com/ETLOnline/spark into fearure/issue-220-adpater-s3-azure
